cmd/cosign/cli/attest/attest_blob.go

```diff
@@ -165,9 +165,35 @@ func (c *AttestBlobCommand) Exec(ctx context.Context, artifactPath string) error
 	var rekorEntry *models.LogEntryAnon
 
 	if c.TSAServerURL != "" {
-		timestampBytes, err = tsa.GetTimestampedSignature(sig, client.NewTSAClient(c.TSAServerURL))
-		if err != nil {
-			return err
+		// We need to decide what signature to send to the timestamp authority.
+		//
+		// Historically, cosign sent `sig`, which is the entire JSON DSSE
+		// Envelope. However, when sigstore clients are verifying a bundle they
+		// will use the DSSE Sig field, so we choose what signature to send to
+		// the timestamp authority based on our output format.
+		if c.NewBundleFormat {
```

Comment:
Can we add a test under e2e_test as well? A copy of https://github.com/sigstore/cosign/blob/081dea1918e9536c1fe233aa2596301381967b3b/test/e2e_test.go#L771 that uses AttestBlobCommand?

Reply:
Sure! This would have been a lot harder if I had not written https://github.com/sigstore/cosign/pull/3876 recently 😅